service/collector/deployment.go

```diff
@@ -102,43 +105,74 @@ func (d *Deployment) Collect(ch chan<- prometheus.Metric) error {
 				return microerror.Mask(err)
 			}
 
+			clusterID := key.ClusterID(customObject)
+
 			{
 				deploymentsClient, err := d.getDeploymentsClient(customObject)
 				if err != nil {
 					return microerror.Mask(err)
 				}
 
-				r, err := deploymentsClient.ListByResourceGroup(context.Background(), key.ClusterID(customObject), "", to.Int32Ptr(100))
+				r, err := deploymentsClient.ListByResourceGroup(context.Background(), clusterID, "", to.Int32Ptr(100))
```

Comment:
Using [`ListByResourceGroupComplete`](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources#Client.ListByResourceGroupComplete) allow to write one for loop less.

```go
r, _ := deploymentsClient.ListByResourceGroup(...)
for r.NotDone() {
    v := range r.Value()

    ...
}
```